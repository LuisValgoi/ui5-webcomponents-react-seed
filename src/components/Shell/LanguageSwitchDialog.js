import React from 'react';
import { useTranslation } from 'react-i18next';

import { Option } from '@ui5/webcomponents-react/lib/Option';
import { Select } from '@ui5/webcomponents-react/lib/Select';
import { setTheme } from '@ui5/webcomponents-base/dist/config/Theme.js';
import InformationDialog from '../InformationDialog/InformationDialog';

const LanguageSwitchDialog = ({ dialogRef }) => {
  const { t } = useTranslation();
  let selectedTheme = localStorage.getItem('reactSeedSelectedTheme');
  setTheme(selectedTheme);

  const onChange = (event) => {
    selectedTheme = event.detail.selectedOption.dataset.value;
    localStorage.setItem('reactSeedSelectedTheme', selectedTheme);
    setTheme(selectedTheme);
  };
  const themeOptions = [
    { value: 'sap_fiori_3', title: t('shell.button.user.settings.item.themeSwitch.option.default') },
    { value: 'sap_fiori_3_dark', title: t('shell.button.user.settings.item.themeSwitch.option.dark') },
    { value: 'sap_belize', title: t('shell.button.user.settings.item.themeSwitch.option.belize') },
    { value: 'sap_fiori_3_hcb', title: t('shell.button.user.settings.item.themeSwitch.option.highContrastBlack') },
    { value: 'sap_fiori_3_hcw', title: t('shell.button.user.settings.item.themeSwitch.option.highContrastWhite') },
  ];
  const style = {
    select: {
      width: '100%',
    },
  };

  return (
    <InformationDialog dialogRef={dialogRef} headerText={t('shell.button.user.settings.item.themeSwitch.title')}>
      <Select onChange={onChange} style={style.select}>
        {themeOptions &&
          themeOptions.map((option) => {
            return (
              <Option key={option.value} data-value={option.value} selected={option.value === selectedTheme}>
                {option.title}
              </Option>
            );
          })}
      </Select>
    </InformationDialog>
  );
};

export default LanguageSwitchDialog;
