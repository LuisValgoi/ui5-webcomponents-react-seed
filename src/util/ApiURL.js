export default {
  GET_USER_LOGGED: '/v1/user/logged',
  GET_TODO_LIST: '/v1/todo/all',
  GET_TODO_BY_ID: '/v1/todo/:id'
};
