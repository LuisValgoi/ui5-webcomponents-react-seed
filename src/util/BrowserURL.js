export default {
  HOME: '/',
  TODO_LIST: '/todo/list',
  TODO_DETAIL: '/todo/:id/detail',
  NOT_FOUND: '/notFound',
  BUGGY: '/buggy',
  ANY: '/*'
};
