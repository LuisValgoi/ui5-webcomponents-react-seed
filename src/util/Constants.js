export default {
  REACT_QUERY: {
    CODES: {
      SUCCESS: 'success',
      LOADING: 'loading',
      ERROR: 'error',
    },
    KEYS: {
      RQ_GET_USER_LOGGED: 'RQ_GET_USER_LOGGED',
      RQ_GET_TODO_LIST: 'RQ_GET_TODO_LIST',
      GET_TODO_BY_ID: 'GET_TODO_BY_ID',
    },
  },
  SEED: {
    SELECTED_THEME: 'SELECTED_THEME',
  },
};
