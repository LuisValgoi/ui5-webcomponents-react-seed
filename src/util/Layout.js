import React from 'react';
import { Text } from '@ui5/webcomponents-react/lib/Text';

export const TextWrapper = ({ children, isOnHeader, style }) => {
  return (
    <Text
      tooltip={children}
      wrapping
      className={isOnHeader && 'text-header-color'}
      style={{ ...style, lineHeight: '20px' }}
    >
      {children}
    </Text>
  );
};
