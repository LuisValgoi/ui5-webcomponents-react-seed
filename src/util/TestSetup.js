import React from 'react';

import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import BrowserProvider from './browser/BrowserProvider';

const render = (ui, { route = BrowserProvider.getUrl('HOME'), ...renderOptions } = {}) => {
  const WrapperProvider = ({ ...props }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <BrowserRouter>
        <Router history={history}>{props.children}</Router>
      </BrowserRouter>
    );
  };

  return rtlRender(ui, { wrapper: WrapperProvider, ...renderOptions });
};

const serverCustom = (apiUrl, data) => {
  return setupServer(
    rest.get(`*${apiUrl}`, (req, res, ctx) => {
      return res(ctx.json(data));
    }),
  );
};

export * from '@testing-library/react';

export { render, serverCustom };
