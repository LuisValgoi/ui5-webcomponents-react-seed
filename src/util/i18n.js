import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import LanguageDetector from 'i18next-browser-languagedetector';

import translationEN from '../locales/en/translation.json';
import translationPT from '../locales/pt/translation.json';

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    keySeparator: false,
    interpolation: {
      escapeValue: false,
    },
    resources: {
      en: {
        translation: translationEN,
      },
      pt: {
        translation: translationPT,
      },
    },
    react: {
      useSuspense: false,
    },
  });

export default i18n;
