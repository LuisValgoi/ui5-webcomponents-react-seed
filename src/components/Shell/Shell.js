import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { ShellBar } from '@ui5/webcomponents-react/lib/ShellBar';
import BrowserURL from '../../util/BrowserURL';

const style = {
  shell: {
    position: 'fixed',
    width: '100%',
    zIndex: 100
  }
};

const Shell = ({ title, ...props }) => {
  const { t } = useTranslation();
  const history = useHistory();

  return (
    <ShellBar
      data-testid='shell-wrapper'
      style={style.shell}
      onLogoClick={() => history.push(BrowserURL.HOME)}
      primaryTitle={title}
      logo={
        <img alt={t('shell.logo.alt')} src="https://sap.github.io/ui5-webcomponents/assets/images/sap-logo-svg.svg" />
      }
      {...props} />
  );
}

export default Shell;
