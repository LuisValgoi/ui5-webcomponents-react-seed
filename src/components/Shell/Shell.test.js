import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../util/TestSetup';

import Shell from './Shell';

describe('Shell.js Test Suite', () => {
  beforeEach(() => {
    render(<Shell />);
  });
  test('should render', () => {
    const shell = screen.getByTestId('shell-wrapper');

    expect(shell).toBeInTheDocument();
  });

  test('should contain an empty space after the Shell Element', () => {
    const emptySpace = screen.getByTestId('emptySpace-wrapper');

    expect(emptySpace).toBeInTheDocument();
  });

  test('should contain an profile element', () => {
    const { container } = render(<Shell title="Test" />);
    const profile = container.querySelector('[slot="profile"]');

    expect(profile).toBeInTheDocument();
  });

  test('should render with primaryTitle "Test"', () => {
    render(<Shell title="Test" />);

    const shell = screen.queryAllByTestId('shell-wrapper')[1];
    expect(shell).toBeInTheDocument();
    expect(shell).toHaveProperty('primaryTitle', 'Test');
  });
});
