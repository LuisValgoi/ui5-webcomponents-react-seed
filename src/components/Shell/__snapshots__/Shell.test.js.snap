// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <ui5-shellbar
    data-testid="shell-wrapper"
    style="position: fixed; width: 100%; z-index: 100;"
  >
    <img
      alt="shell.logo.alt"
      slot="logo"
      src="https://sap.github.io/ui5-webcomponents/assets/images/sap-logo-svg.svg"
    />
  </ui5-shellbar>
</DocumentFragment>
`;
